sycl/source/detail/memory_manager.cpp

```diff
@@ -137,85 +48,25 @@ void memBufferCreateHelper(const AdapterPtr &Adapter, ur_context_handle_t Ctx,
-    xpti::utils::finally _{[&] {
-      // C-style cast is required for MSVC
-      uintptr_t MemObjID = (uintptr_t)(*RetMem);
-      ur_native_handle_t Ptr = 0;
-      // Always use call_nocheck here, because call may throw an exception,
-      // and this lambda will be called from destructor, which in combination
-      // rewards us with UB.
-      // When doing buffer interop we don't know what device the memory should
-      // be resident on, so pass nullptr for Device param. Buffer interop may
-      // not be supported by all backends.
-      Adapter->call_nocheck<UrApiKind::urMemGetNativeHandle>(
-          *RetMem, /*Dev*/ nullptr, &Ptr);
-      emitMemAllocEndTrace(MemObjID, (uintptr_t)(Ptr), Size, 0 /* guard zone */,
-                           CorrID);
-    }};
-#endif
-    if (Size)
-      Adapter->call<UrApiKind::urMemBufferCreate>(Ctx, Flags, Size, Props,
-                                                  RetMem);
-  }
+  if (Size)
+    Adapter->call<UrApiKind::urMemBufferCreate>(Ctx, Flags, Size, Props,
+                                                RetMem);
 }
 
 void memReleaseHelper(const AdapterPtr &Adapter, ur_mem_handle_t Mem) {
   // FIXME urMemRelease does not guarante memory release. It is only true if
   // reference counter is 1. However, SYCL runtime currently only calls
   // urMemRetain only for OpenCL interop
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  uint64_t CorrID = 0;
-  // C-style cast is required for MSVC
-  uintptr_t MemObjID = (uintptr_t)(Mem);
-  uintptr_t Ptr = 0;
-  // Do not make unnecessary UR calls without instrumentation enabled
-  if (xptiTraceEnabled()) {
-    ur_native_handle_t PtrHandle = 0;
-    // When doing buffer interop we don't know what device the memory should be
-    // resident on, so pass nullptr for Device param. Buffer interop may not be
-    // supported by all backends.
-    Adapter->call_nocheck<UrApiKind::urMemGetNativeHandle>(Mem, /*Dev*/ nullptr,
-                                                           &PtrHandle);
-    Ptr = (uintptr_t)(PtrHandle);
-  }
-#endif
-  // We only want to instrument urMemRelease
-  {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-    CorrID = emitMemReleaseBeginTrace(MemObjID, Ptr);
-    xpti::utils::finally _{
-        [&] { emitMemReleaseEndTrace(MemObjID, Ptr, CorrID); }};
-#endif
-    Adapter->call<UrApiKind::urMemRelease>(Mem);
-  }
+  Adapter->call<UrApiKind::urMemRelease>(Mem);
 }
 
 void memBufferMapHelper(const AdapterPtr &Adapter, ur_queue_handle_t Queue,
                         ur_mem_handle_t Buffer, bool Blocking,
                         ur_map_flags_t Flags, size_t Offset, size_t Size,
                         uint32_t NumEvents, const ur_event_handle_t *WaitList,
                         ur_event_handle_t *Event, void **RetMap) {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  uint64_t CorrID = 0;
-  uintptr_t MemObjID = (uintptr_t)(Buffer);
-#endif
   // We only want to instrument urEnqueueMemBufferMap
```

Comment:
not really clear what this comment relates to but probably it is better to remove it also.